lib/lib-dynamodb-data-mapper/README.md

```diff
@@ -1,132 +1,471 @@
 
-⸻
+## Getting started
 
-## baseCommands/
+> Work in progress! The types and interface are subject to change as implementation and review progress.
 
-• Optional base class or shared logic reused across commands
-• Good for things like schema marshalling/unmarshalling, base command input types
+### Installation
 
-## commands/
+```bash
+npm install @aws-sdk/lib-dynamodb-data-mapper
+```
 
-• Contains individual implementations: PutCommand.ts, GetCommand.ts, etc.
-• Each file encapsulates a single logical operation (like SDK Command classes)
+> Other dependencies:
 
-## pagination/
+```bash
+npm install @aws-sdk/client-dynamodb
+```
 
-• Will contain async iterator helpers for query and scan
-• Example: paginateQuery.ts, paginateScan.ts
+### DataMapper public interface
+
+> Subject to change due to review and implementation in progress
+
+```typescript
+/**
+ * Configuration options for initializing a {@link DataMapper}.
+ *
+ * @public
+ */
+export interface DataMapperConfig {
+  /**
+   * The underlying DynamoDB client instance used to execute operations.
+   */
+  client: DynamoDBClient;
+
+  /**
+   * Optional configuration for customizing how data is marshalled into
+   * DynamoDB AttributeValue format.
+   *
+   * See {@link marshallOptions} for supported options such as:
+   * - `convertEmptyValues`: Convert empty strings, blobs, and sets to `null`
+   * - `removeUndefinedValues`: Exclude undefined values from the marshalled output
+   */
+  translateConfig?: marshallOptions;
+}
 
-## decorators/
+/**
+ * @public
+ * Interface for a schema-aware DataMapper supporting both low-level and high-level operations.
+ *
+ * High-level methods operate on domain objects with schema metadata (via decorators or static symbols).
+ * Low-level methods accept raw AWS SDK command inputs for advanced use cases.
+ */
+export class DataMapper {
+  /**
+   * Save a domain model object to DynamoDB using its schema metadata.
+   *
+   * @param item - The model instance with schema/table metadata.
+   * @param criteria - Optional conditions to apply to the put operation (excluding TableName).
+   * @param options - Optional HTTP handler options.
+   * (e.g., timeouts, retries, abort signals). See {@link HttpHandlerOptions}.
+   * @returns The saved item.
+   */
+  put<T extends object>(
+    item: T,
+    criteria?: Omit<Partial<PutItemCommandInput>, "TableName" | "Item">,
+    options?: HttpHandlerOptions
+  ): Promise<T>;
+
+  /**
+   * Retrieve and hydrate a domain object using schema metadata.
+   *
+   * @param key - The primary key of the item.
+   * @param modelCtor - The model class constructor.
+   * @param options - Optional HTTP handler options.
+   * (e.g., timeouts, retries, abort signals). See {@link HttpHandlerOptions}.
+   * @returns A hydrated instance of the model.
+   */
+  get<T extends object>(key: Partial<T>, modelCtor: new () => T, options?: HttpHandlerOptions): Promise<T | undefined>;
```

Comment:
`new () =>. T` requires a class have a no-args constructor. This isn't very flexible. Are you sure this is how users want to interact with the library?

It should accept any factory function of the signature 
```ts
<D, T>(data: D) => T;
```